File: src/datos/repository/EstudianteRepository.js
export class EstudianteRepository {
  constructor() {
    
    if (EstudianteRepository.instance) {
      return EstudianteRepository.instance;
    }
        
    this.estudiantes = [];
    EstudianteRepository.instance = this;
  }
  static getInstance() {
    if (!EstudianteRepository.instance) {
      EstudianteRepository.instance = new EstudianteRepository();
    }
    return EstudianteRepository.instance;
  }
  agregar(estudiante) {
    this.estudiantes.push(estudiante);
  }

  editar(id, nuevosDatos) {
    const index = this.estudiantes.findIndex(e => e.id === id);
    if (index === -1) return false;
    this.estudiantes[index] = { ...this.estudiantes[index], ...nuevosDatos };
    return true;
  }

  eliminar(id) {
      const originalLength = this.estudiantes.length;
      this.estudiantes = this.estudiantes.filter(e => e.id !== id);
    return this.estudiantes.length < originalLength;
  }

  listar() {
    return this.estudiantes;
  }

  buscarPorId(id) {
    return this.estudiantes.find(e => e.id === id);
  }
}

File: crud_builder/src/main/java/ec/edu/espe/datos/repository/EstudianteRepository.java
package ec.edu.espe.datos.repository;

import ec.edu.espe.datos.model.Estudiante;
import java.util.ArrayList;
import java.util.List;

public class EstudianteRepository {

    private static EstudianteRepository instance;

    private List<Estudiante> estudiantes;

    private EstudianteRepository() {
        estudiantes = new ArrayList<>();
    }

    public static EstudianteRepository getInstance() {
        if (instance == null) {
            instance = new EstudianteRepository();
        }
        return instance;
    }

    public void agregar(Estudiante estudiante) {
        estudiantes.add(estudiante);
    }

    public boolean editar(String id, String nombres, int edad) {
        Estudiante est = buscarPorId(id);
        if (est == null) return false;

        est.setNombres(nombres);
        est.setEdad(edad);
        return true;
    }

    public boolean eliminar(String id) {
        return estudiantes.removeIf(e -> e.getId().equals(id));
    }

    public List<Estudiante> listar() {
        return estudiantes;
    }

    public Estudiante buscarPorId(String id) {
        return estudiantes.stream()
                .filter(e -> e.getId().equals(id))
                .findFirst()
                .orElse(null);
    }
}
